### PR TITLE
Add new userinfo method for auth endpoints

### PR DESCRIPTION
**Pull request by joshcanhelp**
- Add new `userinfo` method for `Auth0::Api::AuthenticationEndpoints`
- Add the ability to include additional headers in `get` method in `Auth0::Mixins::HTTPProxy`
- Add note to deprecate previous `user_info` method

### CODE REVIEW

File: lib/auth0/mixins/httpproxy.rb [hunk 1]
```diff
@@ -7,11 +7,12 @@ module HTTPProxy
 
       # proxying requests from instance methods to HTTP class methods
       %i(get post post_file put patch delete).each do |method|
-        define_method(method) do |path, body = {}|
+        define_method(method) do |path, body = {}, extra_headers = {}|
           safe_path = URI.escape(path)
           body = body.delete_if { |_, v| v.nil? }
           result = if [:get, :delete].include?(method)
-                     call(method, url(safe_path), timeout, add_headers(params: body))
+                     add_headers(extra_headers.nil? ? {params: body} : extra_headers )
```
Comment (machuga) [inline lib/auth0/mixins/httpproxy.rb]: Is this usage of `add_headers` compatible with the existing one? Does `add_headers` return a value or does it mutate something?

File: spec/lib/auth0/api/authentication_endpoints_spec.rb [hunk 2]
```diff
@@ -330,6 +330,14 @@
     end
   end
 
+  context '.userinfo' do
+    it { expect(@instance).to respond_to(:userinfo) }
+    it 'is expected to make a GET request to /userinfo' do
+      expect(@instance).to receive(:get).with('/userinfo', nil, {Authorization: 'Bearer access-token'})
+      @instance.userinfo 'access-token'
```
Comment (machuga) [inline spec/lib/auth0/api/authentication_endpoints_spec.rb]: What is this doing?

File: spec/lib/auth0/api/authentication_endpoints_spec.rb [hunk 3]
```diff
@@ -330,6 +330,14 @@
     end
   end
 
+  context '.userinfo' do
+    it { expect(@instance).to respond_to(:userinfo) }
```
Comment (machuga) [inline spec/lib/auth0/api/authentication_endpoints_spec.rb]: If you setup something like `subject { @instance }`, you could even shorten this to `it { is_expected.to respond_to(:user_info) }`, I believe. Not suggesting you need to do this, just showing you some fun RSpec stuff 👍

File: lib/auth0/mixins/httpproxy.rb (hunk 1 above)
Comment (joshcanhelp) [reply lib/auth0/mixins/httpproxy.rb]: > Is this usage of `add_headers` compatible with the existing one? 

Yes. That's there to ... add headers but the current implementation uses it to add URL params by setting `:params` to be the `body` that's passed in for `GET` requests. It's a bit confusing in it's current state but I think this is a good compromise between necessary functionality and legacy support. 

> Does `add_headers` return a value or does it mutate something?

Both. It mutates the module's `@headers` property and returns it. I'm using it just to mutate, then pass the property to that `call()` method.

**Comment by joshcanhelp** [comment]
@machuga - Added integration tests and swapped out the new Client ID. Removed Client Secret and API Token out of the recordings 👍

File: spec/fixtures/vcr_cassettes/Auth0_Api_AuthenticationEndpoints/_change_password/should_trigger_a_password_reset.yml [hunk 4]
```diff
@@ -5,22 +5,22 @@ http_interactions:
...
       - gzip, deflate
       User-Agent:
-      - Ruby/2.3.1
+      - Ruby/2.5.1
       Content-Type:
       - application/json
       Auth0-Client:
       - eyJuYW1lIjoicnVieS1hdXRoMCIsInZlcnNpb24iOiI0LjUuMCJ9
       Authorization:
-      - Bearer eyJ0eXAiOiJKV1QiLCJhbGc.eyJpc3MiOiJodHRwczovL2F1dGgwLXNkay10ZXN0cy5hdXRoMC5jb20vIiwic3ViIjoiQjRvUEhsMDA3VmExR0JkWlhpU0hhRUNVcVZXa2Q5WGtAY2xpZW50cyIsIm.PxfZjVd4wTb_bFbSTENdTmbj13CDQaPK352w3UNL3i3DnWcVJa6qSiA0hCr_tSU_uC34mHkK52O8tFVeZjxCYSeM9yOZUcKVya5N0I7G90X
```
Comment (joshcanhelp) [inline spec/fixtures/vcr_cassettes/Auth0_Api_AuthenticationEndpoints/_change_password/should_trigger_a_password_reset.yml]: Dummy API token, replacing with something more clearly fake

File: spec/fixtures/vcr_cassettes/Auth0_Api_AuthenticationEndpoints/_saml_metadata/should_retrieve_SAML_metadata.yml [hunk 5]
```diff
@@ -12,13 +12,13 @@ http_interactions:
...
       - gzip, deflate
       User-Agent:
-      - Ruby/2.3.1
+      - Ruby/2.5.1
       Content-Type:
       - application/json
       Auth0-Client:
       - eyJuYW1lIjoicnVieS1hdXRoMCIsInZlcnNpb24iOiI0LjUuMCJ9
       Authorization:
-      - Bearer eyJ0eXAiOiJKV1QiLCJhbGc.eyJpc3MiOiJodHRwczovL2F1dGgwLXNkay10ZXN0cy5hdXRoMC5jb20vIiwic3ViIjoiQjRvUEhsMDA3VmExR0JkWlhpU0hhRUNVcVZXa2Q5WGtAY2xpZW50cyIsIm.PxfZjVd4wTb_bFbSTENdTmbj13CDQaPK352w3UNL3i3DnWcVJa6qSiA0hCr_tSU_uC34mHkK52O8tFVeZjxCYSeM9yOZUcKVya5N0I7G90X
```
Comment (joshcanhelp) [inline spec/fixtures/vcr_cassettes/Auth0_Api_AuthenticationEndpoints/_saml_metadata/should_retrieve_SAML_metadata.yml]: Dummy API token, replacing with something more clearly fake

File: spec/fixtures/vcr_cassettes/Auth0_Api_AuthenticationEndpoints/_wsfed_metadata/should_retrieve_WSFED_metadata.yml [hunk 6]
```diff
@@ -12,13 +12,13 @@ http_interactions:
...
       - gzip, deflate
       User-Agent:
-      - Ruby/2.3.1
+      - Ruby/2.5.1
       Content-Type:
       - application/json
       Auth0-Client:
       - eyJuYW1lIjoicnVieS1hdXRoMCIsInZlcnNpb24iOiI0LjUuMCJ9
       Authorization:
-      - Bearer eyJ0eXAiOiJKV1QiLCJhbGc.eyJpc3MiOiJodHRwczovL2F1dGgwLXNkay10ZXN0cy5hdXRoMC5jb20vIiwic3ViIjoiQjRvUEhsMDA3VmExR0JkWlhpU0hhRUNVcVZXa2Q5WGtAY2xpZW50cyIsIm.PxfZjVd4wTb_bFbSTENdTmbj13CDQaPK352w3UNL3i3DnWcVJa6qSiA0hCr_tSU_uC34mHkK52O8tFVeZjxCYSeM9yOZUcKVya5N0I7G90X
```
Comment (joshcanhelp) [inline spec/fixtures/vcr_cassettes/Auth0_Api_AuthenticationEndpoints/_wsfed_metadata/should_retrieve_WSFED_metadata.yml]: Dummy API token, replacing with something more clearly fake

File: spec/fixtures/vcr_cassettes/Auth0_Api_AuthenticationEndpoints/create_test_user.yml [hunk 7]
```diff
@@ -5,22 +5,22 @@ http_interactions:
...
       - gzip, deflate
       User-Agent:
-      - Ruby/2.3.1
+      - Ruby/2.5.1
       Content-Type:
       - application/json
       Auth0-Client:
       - eyJuYW1lIjoicnVieS1hdXRoMCIsInZlcnNpb24iOiI0LjUuMCJ9
       Authorization:
-      - Bearer eyJ0eXAiOiJKV1QiLCJhbGc.eyJpc3MiOiJodHRwczovL2F1dGgwLXNkay10ZXN0cy5hdXRoMC5jb20vIiwic3ViIjoiQjRvUEhsMDA3VmExR0JkWlhpU0hhRUNVcVZXa2Q5WGtAY2xpZW50cyIsIm.PxfZjVd4wTb_bFbSTENdTmbj13CDQaPK352w3UNL3i3DnWcVJa6qSiA0hCr_tSU_uC34mHkK52O8tFVeZjxCYSeM9yOZUcKVya5N0I7G90X
```
Comment (joshcanhelp) [inline spec/fixtures/vcr_cassettes/Auth0_Api_AuthenticationEndpoints/create_test_user.yml]: Dummy API token, replacing with something more clearly fake

File: spec/fixtures/vcr_cassettes/Auth0_Api_AuthenticationEndpoints/delete_test_user.yml [hunk 8]
```diff
@@ -12,13 +12,13 @@ http_interactions:
...
       - gzip, deflate
       User-Agent:
-      - Ruby/2.3.1
+      - Ruby/2.5.1
       Content-Type:
       - application/json
       Auth0-Client:
       - eyJuYW1lIjoicnVieS1hdXRoMCIsInZlcnNpb24iOiI0LjUuMCJ9
       Authorization:
-      - Bearer eyJ0eXAiOiJKV1QiLCJhbGc.eyJpc3MiOiJodHRwczovL2F1dGgwLXNkay10ZXN0cy5hdXRoMC5jb20vIiwic3ViIjoiQjRvUEhsMDA3VmExR0JkWlhpU0hhRUNVcVZXa2Q5WGtAY2xpZW50cyIsIm.PxfZjVd4wTb_bFbSTENdTmbj13CDQaPK352w3UNL3i3DnWcVJa6qSiA0hCr_tSU_uC34mHkK52O8tFVeZjxCYSeM9yOZUcKVya5N0I7G90X
```
Comment (joshcanhelp) [inline spec/fixtures/vcr_cassettes/Auth0_Api_AuthenticationEndpoints/delete_test_user.yml]: Dummy API token, replacing with something more clearly fake

File: spec/fixtures/vcr_cassettes/Auth0_Api_V2_Blacklists/_add_token_to_blacklist/should_add_a_token_to_the_blacklist.yml [hunk 9]
```diff
@@ -5,20 +5,20 @@ http_interactions:
...
       - gzip, deflate
       User-Agent:
-      - Ruby/2.3.1
+      - Ruby/2.5.1
       Content-Type:
       - application/json
       Auth0-Client:
       - eyJuYW1lIjoicnVieS1hdXRoMCIsInZlcnNpb24iOiI0LjUuMCJ9
       Authorization:
-      - Bearer eyJ0eXAiOiJKV1QiLCJhbGc.eyJpc3MiOiJodHRwczovL2F1dGgwLXNkay10ZXN0cy5hdXRoMC5jb20vIiwic3ViIjoiQjRvUEhsMDA3VmExR0JkWlhpU0hhRUNVcVZXa2Q5WGtAY2xpZW50cyIsIm.PxfZjVd4wTb_bFbSTENdTmbj13CDQaPK352w3UNL3i3DnWcVJa6qSiA0hCr_tSU_uC34mHkK52O8tFVeZjxCYSeM9yOZUcKVya5N0I7G90X
```
Comment (joshcanhelp) [inline spec/fixtures/vcr_cassettes/Auth0_Api_V2_Blacklists/_add_token_to_blacklist/should_add_a_token_to_the_blacklist.yml]: Dummy API token, replacing with something more clearly fake

File: spec/fixtures/vcr_cassettes/Auth0_Api_V2_Blacklists/_blacklisted_tokens/should_get_the_added_token_from_the_blacklist.yml [hunk 10]
```diff
@@ -12,13 +12,13 @@ http_interactions:
...
       - gzip, deflate
       User-Agent:
-      - Ruby/2.3.1
+      - Ruby/2.5.1
       Content-Type:
       - application/json
       Auth0-Client:
       - eyJuYW1lIjoicnVieS1hdXRoMCIsInZlcnNpb24iOiI0LjUuMCJ9
       Authorization:
-      - Bearer eyJ0eXAiOiJKV1QiLCJhbGc.eyJpc3MiOiJodHRwczovL2F1dGgwLXNkay10ZXN0cy5hdXRoMC5jb20vIiwic3ViIjoiQjRvUEhsMDA3VmExR0JkWlhpU0hhRUNVcVZXa2Q5WGtAY2xpZW50cyIsIm.PxfZjVd4wTb_bFbSTENdTmbj13CDQaPK352w3UNL3i3DnWcVJa6qSiA0hCr_tSU_uC34mHkK52O8tFVeZjxCYSeM9yOZUcKVya5N0I7G90X
```
Comment (joshcanhelp) [inline spec/fixtures/vcr_cassettes/Auth0_Api_V2_Blacklists/_blacklisted_tokens/should_get_the_added_token_from_the_blacklist.yml]: Dummy API token, replacing with something more clearly fake

File: spec/spec_helper.rb [hunk 11]
```diff
@@ -6,9 +6,13 @@
 
 require 'vcr'
 VCR.configure do |config|
+  # config.allow_http_connections_when_no_cassette = true
   config.cassette_library_dir = 'spec/fixtures/vcr_cassettes'
   config.configure_rspec_metadata!
   config.hook_into :webmock
+  config.filter_sensitive_data('CLIENT_SECRET') { ENV['CLIENT_SECRET'] }
```
Comment (joshcanhelp) [inline spec/spec_helper.rb]: 🎉
